samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -52,6 +52,9 @@
   public static final boolean DEFAULT_DISALLOW_LARGE_MESSAGES = false;
   public static final String DROP_LARGE_MESSAGES = STORE_PREFIX + "%s.drop.large.messages";
   public static final boolean DEFAULT_DROP_LARGE_MESSAGES = false;
+  // The log compaction lag time for transactional state change log
+  public static final String CHANGELOG_MIN_COMPACTION_LAG_MS = STORE_PREFIX + "%s.changelog.kafka.min.compaction.lag.ms";
```

Comment:
Should this be 'changelog.min.compaction.lag.ms' to match the other Kafka related properties above (replication.factor and max.message.size.bytes)?

Reply:
I think we are in an inconsistent position for this config. On one hand, the logic in KafkaConfig.getChangelogKafkaProperties() already allows overriding any kafka properties by using stores.x.changelog.kafka..... So the user can already override things like replication.factor using stores.x.changelog.kafka.replication.factor. The changelog configs in StorageConfig make a second way to configure it, which is problematic since the user can use the original kafka override to do it and StorageConfig code getters won't return the correct value. Seems like an existing bug to me already.

Not sure how many use cases are using those two configs, but it's problematic in the long run.

Reply:
Oh I see, that's quite confusing. Even the configuration reference documents the changelog.kafka.* properties incorrectly (provided example is incorrect) and does not clarify its precedence. Looks like the convention is: if it's a known property that the framework cares about, set it explicitly using the changelog.xyz convention. If it's an arbitrary Kafka property, set it using changelog.kafka.xyz convention. 

Following this convention, and for consistency with existing behavior class, I'd still prefer to rename this to changelog.min.compaction.lag.ms. 

Seems like long term fix would be to give higher precedence to the explicitly specified properties over the kafka.* specified properties, although that's backwards incompatible. We can create a ticket to track that separately.